group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroupMember.java

```diff
@@ -0,0 +1,756 @@
+        }
+
+        public StreamsGroupMember build() {
+            return new StreamsGroupMember(
+                memberId,
+                memberEpoch,
+                previousMemberEpoch,
+                instanceId,
+                rackId,
+                rebalanceTimeoutMs,
+                clientId,
+                clientHost,
+                topologyEpoch,
+                processId,
+                userEndpoint,
+                clientTags,
+                state,
+                assignedActiveTasks,
+                assignedStandbyTasks,
+                assignedWarmupTasks,
+                activeTasksPendingRevocation,
+                standbyTasksPendingRevocation,
+                warmupTasksPendingRevocation
+            );
+        }
+    }
+
+    /**
+     * The member id.
+     */
+    private final String memberId;
+
+    /**
+     * The current member epoch.
+     */
+    private final int memberEpoch;
+
+    /**
+     * The previous member epoch.
+     */
+    private final int previousMemberEpoch;
+
+    /**
+     * The member state.
+     */
+    private final MemberState state;
+
+    /**
+     * The instance id provided by the member.
+     */
+    private final String instanceId;
+
+    /**
+     * The rack id provided by the member.
+     */
+    private final String rackId;
+
+    /**
+     * The rebalance timeout provided by the member.
+     */
+    private final int rebalanceTimeoutMs;
+
+    /**
+     * The client id reported by the member.
+     */
+    private final String clientId;
+
+    /**
+     * The host reported by the member.
+     */
+    private final String clientHost;
+
+    /**
+     * The topology epoch
+     */
+    private final int topologyEpoch;
+
+    /**
+     * The process ID
+     */
+    private final String processId;
+
+    /**
+     * The endpoint
+     */
+    private final StreamsGroupMemberMetadataValue.Endpoint userEndpoint;
+
+    /**
+     * The client tags
+     */
+    private final Map<String, String> clientTags;
+
+    /**
+     * Active tasks assigned to this member.
+     */
+    private final Map<String, Set<Integer>> assignedActiveTasks;
+
+    /**
+     * Standby tasks assigned to this member.
+     */
+    private final Map<String, Set<Integer>> assignedStandbyTasks;
+
+    /**
+     * Warmup tasks assigned to this member.
+     */
+    private final Map<String, Set<Integer>> assignedWarmupTasks;
+
+    /**
+     * Active tasks being revoked by this member.
+     */
+    private final Map<String, Set<Integer>> activeTasksPendingRevocation;
+
+    /**
+     * Standby tasks being revoked by this member.
+     */
+    private final Map<String, Set<Integer>> standbyTasksPendingRevocation;
+
+    /**
+     * Warmup tasks being revoked by this member.
+     */
+    private final Map<String, Set<Integer>> warmupTasksPendingRevocation;
+
+    @SuppressWarnings("checkstyle:ParameterNumber")
```

Comment:
What are the rules we have around adding suppressions?

Reply:
Add them where it makes sense? 🙂

Reply:
What would be the alternative here?